### DISCUSSION TITLE
Unexpected BGN values

### DISCUSSION THREAD

**Discussion by ThomasLuypaert** [QutEcoacoustics/audio-analysis#509] [Q&A]
I'm writing with a question regarding the BGN-index, which I've computed using the default settings in AnalysisPrograms.

Based on the Towsey (2017) technical report, I was under the impression that amplitude values should be set at -90 dB minimum. However, when computing the BGN-index, I sometimes get values that are far lower, up to - 141 dB, as you can see in the example I attached here below. I also found it interesting that BGN-values seemed to decrease with the frequency bin number - is this a normal pattern you find?

[G9_2_87_20150806_174500Z__Towsey.Acoustic.BGN.zip](https://github.com/QutEcoacoustics/audio-analysis/files/6884101/G9_2_87_20150806_174500Z__Towsey.Acoustic.BGN.zip)

I was wondering if you'd be able to shed some light on what is going on. Am I misinterpreting how the BGN-index is computed based on the technical report, or is something causing this very low values. 

I appreciate your help!

**Comment by towsey** [comment]
Hi Thomas, Thanks for your question and example line in csv file.
First, the easiest one of your questions. Background noise in environmental recordings is close to what is called pink noise. You can look this up, but pink noise means that the power decreases logarithmically with frequency. In your case the BG noise decrease from -101 to -141 dB. This is entirely as expected. Your values are very low for an environmental recording, which suggests to me that this sample is taken from a very quiet place in the middle of winter!

In theory, the smallest possible value for a signal amplitude (ie in the waveform) is determined by the bits per sample. Typically this will be 16 bits. However one of the bits is the sign bit. So the smallest possible amplitude is given in our code by epsilon as follows:
        epsilon => Math.Pow(0.5, BitsPerSample - 1) = 3.05 * 10^-5.

Assuming 16 bit samples (you need to check this is the case for your recordings) the lowest decibel value will be:
        dB = 20 * log10 (epsilon) = 20 * -15 * log10(2) =  -90.3 dB.
In line 53 of the class SNR.cs, I have set a constant:
        public const double MinimumDbBoundForEnvironmentalNoise = -90;

HOWEVER! Note that this constant is only used when calculating the background noise of a waveform signal. See class NoiseRemovalModal.cs, line 119. It is NOT used when dealing with spectrograms.
Just as a matter of interest, when sounds get up to -30 to -20dB that is getting very loud for an environmental recording.

With spectrograms, the situation is more complex. In this case the amplitude spectrogram values are derived from sinusoid coefficients calculated by the FFT algorithm. The minimum accepted value is epsilon^2, but when this is adjusted for the FFT window energy etc, it turns out that the minimum possible decibel value in the decibel spectrogram is -154 dB.  

If you are finding values less than (more negative than) this, then something is happening that I do not understand.

I hope this helps. Cheers, Michael

**Reply by atruskie** [reply]
Some useful links:

- https://en.wikipedia.org/wiki/Pink_noise
- https://en.wikipedia.org/wiki/Decibel

The code snippet @towsey was referring too:

https://github.com/QutEcoacoustics/audio-analysis/blob/45150178b1e363bd00cb2836a092f09b191700df/src/AudioAnalysisTools/DSP/NoiseRemovalModal.cs#L119-L123

Also:

- A decibel unit is a ratio. 
- I think our calculations are done as the ratio of the value to 1.0 unit amplitude.
- See https://en.wikipedia.org/wiki/Decibel and particularly the table on the right
- A value of `-141` would be an amplitude of  `0.0000000891` (`8.91E-08`) which is very quiet
- The smallest representable value in a sample in a 16-bit recording is `3.05E-05` /  `0.000031` /`-90.31 dB`

- There is likely some floating-point computation error here, 
  - but the effect is likely small
  - and as @towsey mentioned above, it seems the FFT can generate smaller values than the recorder could sample. Which puts the values you reported into an expected range.

If you submit a demo file I'd be happy to confirm the values are expected

**Edited**: updated figures, they were incorrect

**Comment by ThomasLuypaert** [comment]
Hi Michael and Anthony!

Thank you for getting back to me so fast, and for the detailed answer. 

Great to hear the decreasing background noise amplitude with frequency is an expected pattern, I'll take a look at the pink noise model in more detail. It is somewhat odd that the background noise values are so low, given that they come from a tropical rain forest in the Brazilian Amazon. Could this be linked to the device which was used to make the recordings?

The device setup was an LG L70 smartphone in a waterproof case with an external omnidirectional microphone recording sound files in the '.wav' format using the Arbimon Touch application. The sampling rate was 44.1 kHz, I'm unfortunately not sure what the bit per sample was, as I did not collect the data myself. I'll see if I can find out more about this. 

@towsey, thank you for clarifying the behind the scenes workings of how the amplitude values are derived at different stages inside the software - as I don't speak C#, this is really useful! I don't think I've found any values below -154 dB, I'll have a quick look to see if that is not the case. @atruskie, thanks for the additional links, it will be useful to brush up my knowledge on these topics a little more :). 

Here is the sound file that was used to obtain the BGN index files in the message above:
[G9_2_87_20150806_174500Z.zip](https://github.com/QutEcoacoustics/audio-analysis/files/6899003/G9_2_87_20150806_174500Z.zip)

I appreciate your help!

Cheers, 
Thomas

**Comment by towsey** [comment]
Good morning, Concerning the location in code where the calculation of the minimum spectral decibels is calculated, it is done at line 59-60 in the class MFCCStuff.cs. THe method is GetLogEnergySpectrogram(). THis is a low level method and the calculation needs to be done late because the calculation depends on the FFT window energy and the sample rate. Both these are user determined in the case of recognizers. 
Concerning your low signal amplitude, the first place I would check is your recording gain. It could be way too low. What are the False Colour spectrograms like? That will give you a first clue as to recording gain.
Cheers, Michael

**Reply by towsey** [reply]
I have just listened to your example recording and there appears to be nothing in it expect perhaps a slight breeze sound. In other words, its a quiet recording because there is nothing making a sound! Still it would be worth checking the recording gain.
